code/game/machinery/cryopod.dm

```diff
@@ -0,0 +1,413 @@
+		/obj/item/areaeditor/blueprints,
+		/obj/item/clothing/head/helmet/space,
+		/obj/item/clothing/suit/space,
+		/obj/item/clothing/suit/armor,
+		/obj/item/defibrillator/compact,
+		/obj/item/reagent_containers/hypospray/cmo,
+		/obj/item/clothing/accessory/medal/gold/captain,
+		/obj/item/clothing/gloves/krav_maga,
+		/obj/item/nullrod,
+		/obj/item/tank/jetpack,
+		/obj/item/documents,
+		/obj/item/nuke_core_container
+	)
+	// These items will NOT be preserved
+	var/static/list/do_not_preserve_items = list (
+		/obj/item/mmi/posibrain
+	)
+
+/obj/machinery/cryopod/Initialize()
+	..()
+	return INITIALIZE_HINT_LATELOAD //Gotta populate the cryopod computer GLOB first
+
+/obj/machinery/cryopod/LateInitialize()
+	update_icon()
+	find_control_computer()
+
+// This is not a good situation
+/obj/machinery/cryopod/Destroy()
+	control_computer = null
+	return ..()
+
+/obj/machinery/cryopod/proc/find_control_computer(urgent = FALSE)
+
+	for(var/cryo_console as anything in GLOB.cryopod_computers)
+		var/obj/machinery/computer/cryopod/console = cryo_console
+		if(get_area(console) == get_area(src))
+			control_computer = console
+			break
+
+	// Don't send messages unless we *need* the computer, and less than five minutes have passed since last time we messaged
+	if(!control_computer && urgent && COOLDOWN_FINISHED(src, last_no_computer_message))
+		COOLDOWN_START(src, last_no_computer_message, 5 MINUTES)
+		log_admin("Cryopod in [get_area(src)] could not find control computer!")
+		message_admins("Cryopod in [get_area(src)] could not find control computer!")
+		last_no_computer_message = world.time
+
+	return control_computer != null
+
+/obj/machinery/cryopod/close_machine(mob/user)
+	if(!control_computer)
+		find_control_computer(TRUE)
+	if((isnull(user) || istype(user)) && state_open && !panel_open)
+		..(user)
+		var/mob/living/mob_occupant = occupant
+		if(mob_occupant && mob_occupant.stat != DEAD)
+			to_chat(occupant, "<span class='boldnotice'>You feel cool air surround you. You go numb as your senses turn inward.</span>")
+		if(mob_occupant.client || !mob_occupant.key) // Self cryos and SSD
+			despawn_world_time = world.time + (time_till_despawn * 0.1) // This gives them 30 seconds
+		else
+			despawn_world_time = world.time + time_till_despawn
+	icon_state = "cryopod"
+
+/obj/machinery/cryopod/open_machine()
+	..()
+	icon_state = "cryopod-open"
+	density = TRUE
+	name = initial(name)
+
+/obj/machinery/cryopod/container_resist_act(mob/living/user)
+	visible_message("<span class='notice'>[occupant] emerges from [src]!</span>",
+		"<span class='notice'>You climb out of [src]!</span>")
+	open_machine()
+
+/obj/machinery/cryopod/relaymove(mob/user)
+	container_resist_act(user)
+
+/obj/machinery/cryopod/process()
+	if(!occupant)
+		return
+
+	var/mob/living/mob_occupant = occupant
+	if(mob_occupant)
+		// Eject dead people
+		if(mob_occupant.stat == DEAD)
+			open_machine()
+
+		if(!(world.time > despawn_world_time))
```

Comment:
```suggestion
		if(world.time <= despawn_world_time)
```

Though, I think this variable should be renamed as `time_to_despawn`, as its not clear right now what world.time its referring to--that or when it initially despawned.